econml/solutions/causal_analysis/_causal_analysis.py

```diff
@@ -1136,10 +1176,116 @@ def _heterogeneity_tree_string(self, Xtest, feature_index, *,
             A (1-alpha)*100% confidence interval is displayed.
         """
 
-        intrp, feature_names, treatment_names, _ = self._tree(False, Xtest, feature_index,
-                                                              max_depth=max_depth,
-                                                              min_samples_leaf=min_samples_leaf,
-                                                              min_impurity_decrease=min_impurity_decrease,
-                                                              alpha=alpha)
-        return intrp.export_graphviz(feature_names=feature_names,
-                                     treatment_names=treatment_names)
+        intrp, feature_names, _, _ = self._tree(False, Xtest, feature_index,
+                                                max_depth=max_depth,
+                                                min_samples_leaf=min_samples_leaf,
+                                                min_impurity_decrease=min_impurity_decrease,
+                                                alpha=alpha)
+
+        def hetero_data(tree, node_id):
+            return {'effect': _sanitize(tree.value[node_id])}
+        return _tree_interpreter_to_dict(intrp, feature_names, hetero_data)
+
+    def individualized_policy(self, Xtest, feature_index, *, n_rows=None, treatment_costs=0, alpha=0.1):
+        """
+        Get individualized treatment policy based on the learned model for a feature, sorted by the predicted effect.
+
+        Parameters
+        ----------
+        Xtest: array-like
+            Features
+        feature_index: int or string
+            Index of the feature to be considered as treatment
+        n_rows: int, optional
+            How many rows to return (all rows by default)
+        treatment_costs: array-like, default 0
+            Cost of treatment, as a scalar value or per-sample
+        alpha: float in [0, 1], default 0.1
+            Confidence level of the confidence intervals
+            A (1-alpha)*100% confidence interval is returned
+
+        Returns
+        -------
+        output: DataFrame
+            Dataframe containing recommended treatment, effect, confidence interval, sorted by effect
+        """
+        result = self._safe_result_index(Xtest, feature_index)
+
+        # get dataframe with all but selected column
+        orig_df = pd.DataFrame(Xtest, columns=self.feature_names_).drop(self.feature_names_[result.feature_index],
+                                                                        axis=1)
+
+        Xtest = result.X_transformer.transform(Xtest)
+        if Xtest.shape[1] == 0:
+            Xtest = None
+
+        effect = result.estimator.const_marginal_effect_inference(Xtest)
+        if result.feature_baseline is None:
+            # apply 10% of a typical treatment for this feature
+            effect.scale(result.treatment_value * 0.1)
```

Comment:
Instead of add a scale function under Inference class, why not learn the effect direction index from const_marginal_effect, and they create two vector T0 as a vector of `treatment_value` and T1 as a vector of either (treatment_value*1.1) or (treatment_value*0.9) depends on the index of increase or decrease, they call effect(X,T0=T0, T1=T1)? it will automatically give you the right point estimate and CI?

Reply:
I think the scale function is generally useful, but I agree that effect could be used here.